src/Adapter/Filesystem/FilesystemCachePool.php

```diff
@@ -151,11 +151,10 @@ protected function storeItemInCache(CacheItemInterface $item, $ttl)
      */
     private function getFilePath($key)
     {
-        if (!preg_match('|^[a-zA-Z0-9_\.! ]+$|', $key)) {
```

Comment:
I think this check should stay here...  I'm sure there are other characters that are not valid, that should be normalized as well, arent there?